src/foam/nanos/auth/UserRegistrationDAO.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * @license
+ * Copyright 2024 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'UserRegistrationDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: 'Final configuration of new user before creation',
+
+  javaImports: [
+    'foam.core.X',
+    'foam.dao.DAO',
+    'foam.dao.ArraySink',
+    'static foam.mlang.MLang.*',
+    'foam.nanos.logger.Loggers',
+    'foam.nanos.theme.Theme',
+    'foam.nanos.theme.Themes',
+    'foam.util.SafetyUtil',
+    'java.util.List'
+  ],
+
+  messages: [
+    { name: 'EMAIL_REQUIRED', message: 'Email required' },
+    { name: 'DUPLICATE_USER', message: 'User with the same username already exists' }
+  ],
+
+  javaCode: `
+    public UserRegistrationDAO(X x, DAO delegate) {
+      setX(x);
+      setDelegate(delegate);
+    }
+  `,
+
+  methods: [
+    {
+      name: 'put_',
+      javaCode: `
+        User user = register(x, (User) obj);
+        return super.put_(getX(), user);
+      `
+    },
+    {
+      name: 'register',
```

Comment:
would be nice to have documentation: describing what this method does

Reply:
doc added to the model.